test/unified_format.py

```diff
@@ -444,7 +515,7 @@ def get_lsid_for_session(self, session_name):
             return self._session_lsids[session_name]
 
 
-binary_types = (Binary, bytes)
+binary_types = (Binary, bytes, uuid.UUID)
```

Comment:
I'm a little concerned this might lead to masking bugs where we expect to be sending (or returning) Binary but actually end up with UUID. Can we remove this now that we use uuid_representation=unspecified?

Reply:
Trying it out

Reply:
Hmm, the issue is actually coming from the expected [event](https://github.com/mongodb/specifications/blob/0e1d31ec088a4da9380cba95eb987db97bc435ed/source/client-side-encryption/tests/unified/createDataKey.json#L121).  We enforce that `_id`s given to `create_data_key` are of the type [UUID](https://github.com/mongodb/mongo-python-driver/blob/06310391185cad2af5310fa75e68d17b1f46522a/pymongo/encryption.py#L229), but the expected event wants a `Binary` `_id`.  I don't see a precedent for altering the events, but it seems like that might be needed here.

Reply:
```python
 [2022/06/29 21:48:53.819] FAIL: test_create_data_key_with_AWS_KMS_provider (test.test_encryption.TestUnifiedCreateDataKey)
 [2022/06/29 21:48:53.819] ----------------------------------------------------------------------
 [2022/06/29 21:48:53.819] Traceback (most recent call last):
...
 [2022/06/29 21:48:53.819]     self.test.assertIsInstance(value, permissible_types)
 [2022/06/29 21:48:53.819] AssertionError: UUID('55c9396c-0d87-4128-93b0-095b806818cb') is not an instance of (<class 'bson.binary.Binary'>, <class 'bytes'>)
```
We should fix these errors by making _KEY_VAULT_OPTS use unspecified UUID representation (the default) instead of STANDARD. The "data_key _id must be a UUID" check will need to be updated to check for Binary with UUID_SUBTYPE too.

Reply:
Done